Localize around

fuentesloic opened the pull request:
- Find your current place and put a red marker (in front of all of them)
- Put marker and information about interesting place from `/statics/test.json`
- Center view and zoom according your current position

vinyll committed:
preparing /around for users localization

fuentesloic committed:
init map

init css to around map with class nammed you to see directly where the user is

init JS to around map with user special marker, class nammed you

clean html to be readable correctly

add a Json database to develop the js around code

install webworker to respond interesting point positions from test.json

fuentesloic committed:
new webworker to run and receive position then create a classic marker

add popup with name of the point

complete popup with set html and prepare place for put data from webworker

add wifi and powerAvability

received wifi and powerAvaibility data

fix the probleme with comment because it's an array

add content into html popup

remove current position sentence - useless

set zoom and center view on the current location

add a css link to trip

overise trips.css for red marker and current position style with z-index

change mostly syntax error from vinyll review

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+})
+
+// users interesting points
+const giveData = new Worker("/static/js/webworker-around.js");
```
I would rename this to `pois`, or `workspots`, or something more related to what it actually is.

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+const giveData = new Worker("/static/js/webworker-around.js");
+
+giveData.onmessage = (informations) => {
+  const [name,wifi,power,comment,lng,lat] = informations.data
```
space after commas

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+  const el = document.createElement('div')
+  el.classList.add('marker')
+
+  let content = "<h2>"+name+"</h2><p>Wifi quality: "+ wifi+"<br>Power available: "+power+"<br>Comments: "+comment+"</p>"
```
You're using ES6, therefore you should use [template strings](https://babeljs.io/learn-es2015/#ecmascript-2015-features-template-strings)

vinyll replied:
Maybe the wifi , power and comments should be in a list (`<li>`) instead of `<p>`

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+      .addTo(map)
+};
+
+giveData.postMessage("info-requested");
```
quote inconsistency here. You should use single quotes like everywhere else.

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+
+  const el = document.createElement('div');
+  el.classList.add('marker')
+  el.classList.add('you')
```
_you_ as a class name is not explicit. I would use something like _current-location_ or _current_ eventually.

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+// user last location
+if(current_location.length) {
+  const popup = new mapboxgl.Popup({offset: [10, -20]})
+      .setText('you')
```
What do you think of a text like "Your current location"?

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+giveData.postMessage("info-requested");
+
+// user last location
+if(current_location.length) {
```
I know I made the error and I'll need to fix it, but JS takes camelCase writing, not underscores in its variable names. it should therefore be called `currentLocation` everywhere in JS.
Python does take underscore.

fuentesloic replied:
I tried to change in the "user" to save the position into a currentLocation but I have all the time an mongoError ... pretty hard to fix it :D
So I don't correct this one, an when you will fix it I push a correct with currentLocation

vinyll commented on static/js/webworker-around.js:
```diff
@@ -0,0 +1,18 @@
+onmessage = _ => {
+  fetch("/static/test.json")
```
single quotes

vinyll commented on static/js/webworker-around.js:
```diff
@@ -0,0 +1,18 @@
+onmessage = _ => {
+  fetch("/static/test.json")
+    .then(response => {
```
shorter version:

```js
then(response => response.json())
```

vinyll commented on static/js/webworker-around.js:
```diff
@@ -0,0 +1,18 @@
+      return response.json();
+    })
+    .then(json => {
+      const items = json;
```
We do not use semi-columns ";" in this project unless very necessary.

vinyll commented on static/js/around-map.js:
```diff
@@ -0,0 +1,46 @@
+// users interesting points
+const giveData = new Worker("/static/js/webworker-around.js");
+
+giveData.onmessage = (informations) => {
```
parenthesis are not required here. For consistency – because they are not used elsewhere in the code when it's a unique parameter – they need to be removed.

vinyll reviewed (changes requested):
Mostly syntax adaptation.
Meaning the code is good!! 👍

vinyll commented on static/css/around-map.css:
```diff
@@ -1,4 +1,4 @@
-.marker.you {
+.marker.current-location {
     background-image: url('/static/images/marker-red.png');
     z-index: 9999;
```
Do not make a z-index that is so huge. I understand it should be over everything, but what if everything else like an alert banner needs to be over? And then a popup window over that banner alert?
Whe end up with ridiculous counts of _9_s.
Instead why don't you just apply a z-index on the markers (eg: `z-index: 2`), and 1 higher on the `.current-location`?

vinyll commented on static/js/around-map.js:
```diff
@@ -7,15 +7,21 @@ const map = new mapboxgl.Map({
+                    <li>Power available: ${power}</li>
+                    <li>Comments: ${comment}</li>
+                  </ul>`
+
```
Looks 👌

vinyll commented on static/js/around-map.js:
```diff
@@ -25,22 +31,22 @@ giveData.onmessage = (informations) => {
+if(currentLocation.length) {
   const popup = new mapboxgl.Popup({offset: [10, -20]})
-      .setText('you')
+      .setText('You current location')
```
_You**r** current location_

vinyll commented on static/js/webworker-around.js:
```diff
@@ -1,18 +1,16 @@
-    .then(response => {
-      return response.json();
-    })
+  fetch('/static/test.json')
```
Now that you're a big boy, maybe is it time for you to read directly from the API… ;)
`fetch('http://158.58.170.155:3030/spots')`
This way you can get rid of the _test.json_ file 😌

vinyll commented on static/js/webworker-around.js:
```diff
@@ -1,18 +1,16 @@
+    .then(response => response.json())
     .then(json => {
-      const items = json;
+      const items = json
```
```js
.then(items => {
  items.forEach(item => {
    …
```

vinyll reviewed (changes requested):
We're getting there! :)

vinyll committed:
proxied webnomad spots in json

replace current_location by currentLocation

change current_location by currentLocation

change z-index logic with regular number

change test data source to an evolutive webnomad pois database

delete test.json 'cause replace by webnomad evolutive src

vinyll reviewed (changes requested):
It's all perfect, you made it! 🍾 

Now a challenge before merging: 25 commits for one single feature is **a lot**.
You may want to use `git rebase -i a71cf5d9^` and edit which commit you want to squash and which you want to pick. Maybe 2 to 4 commits for the frontend would be enough. (see https://www.atlassian.com/git/tutorials/rewriting-history#git-rebase-i for examples)

fuentesloic committed:
create marker with current location from users to find each other

fuentesloic commented:
@vinyll : what do you think now about this "squash" ?

vinyll commented on static/js/webworker-around.js:
```diff
@@ -0,0 +1,16 @@
+        let comment = item.comments[0]
+        let lng = item.position.longitude
+        let lat = item.position.latitude
+        postMessage([name,wifi,power,comment,lng,lat])
```
space after commas

fuentesloic replied:
oups I forget ... I change that right now and I'll do a `git push -f`and a merge :) youhouuuu ^^

vinyll reviewed (approved):
Commits look good. You just got it!  🎉 
Just one notice about commas in JS (just `push -f` here not to make a new commit only for that).

And you're good to merge!
